### PR TITLE
Update get_translated_payload method in ManagementFeature

### PR DESCRIPTION
**Pull request by Vioshim**
## Rationale

The function requires the bot's tree for working accordingly. Not ready for merging in master but rather once user apps merges with the main branch.

## Summary of changes made

It adds bot.tree as parameter for the function

## Checklist

- [x] This PR changes the jishaku module/cog codebase
    - [ ] These changes add new functionality to the module/cog
    - [x] These changes fix an issue or bug in the module/cog
    - [x] I have tested that these changes work on a production bot codebase
    - [x] I have tested these changes against the CI/CD test suite
    - [ ] I have updated the documentation to reflect these changes
- [ ] This PR changes the CI/CD test suite
    - [ ] I have tested my suite changes are well-formed (all tests can be discovered)
    - [ ] These changes adjust existing test cases
    - [ ] These changes add new test cases
- [ ] This PR changes prose (such as the documentation, README or other Markdown/RST documents)
    - [ ] I have proofread my changes for grammar and spelling issues
    - [ ] I have tested that any changes regarding Markdown/RST syntax result in a well formed document

### CODE REVIEW

**Comment by Gorialis** [comment]
The function signature doesn't indicate that this should be necessary. Can you explain why you thought this change was necessary (and ideally include any supporting errors or other info you can about what lead you towards this fix)?

**Comment by Vioshim** [comment]
image

It errors, but now that I look properly into source code, it has to do with how it works for hybrid commands.
https://github.com/Rapptz/discord.py/blob/425edd2e10b9be3d7799c0df0cd1d43a1a34654e/discord/app_commands/transformers.py#L93, as hybrid commands require the tree and the translator. (Therefore this PR breaks normal app commands, so please don't merge)

**Comment by Gorialis** [comment]
I see. Part of how this code works might need to be rewritten to make it support both cases, I haven't really used hybrid commands so I guess I've never run into this. I imagined there might be some need to refine and update the sync procedure anyway since it deviates from how the discord.py public API intends for the sake of better diagnostics, so I'll probably need to make some kind of test harness to make sure I'm not missing things when maintaining it.

**Comment by tuna2134** [comment]
https://github.com/Rapptz/discord.py/blob/425edd2e10b9be3d7799c0df0cd1d43a1a34654e/discord/ext/commands/hybrid.py#L299

**Comment by tuna2134** [comment]
https://github.com/Rapptz/discord.py/blob/425edd2e10b9be3d7799c0df0cd1d43a1a34654e/discord/app_commands/tree.py#L1060

**Comment by tuna2134** [comment]
I am looking tree.sync function, it looks same.
And I have never seen that report

**Comment by pythonmcpi** [comment]
@Gorialis fyi, this issue is only present in the (currently experimental) feature/user_apps branch of discord.py.

This is caused by a change in `app_commands.Command.to_dict`'s signature, where a new `tree` argument was added. [Permalink to the change](https://github.com/Rapptz/discord.py/blame/08b138cc71fea5594c068082ee565a05cde2c7fa/discord/app_commands/commands.py#L776) | [Link to latest feature/user_apps commit](https://github.com/Rapptz/discord.py/blame/feature/user_apps/discord/app_commands/commands.py#L776)

File: jishaku/features/management.py [hunk 1]
```diff
@@ -241,7 +241,7 @@ async def jsk_sync(self, ctx: ContextA, *targets: str):
             )
             translator = getattr(self.bot.tree, 'translator', None)
             if translator:
-                payload = [await command.get_translated_payload(translator) for command in slash_commands]
+                payload = [await command.get_translated_payload(self.bot.tree, translator) for command in slash_commands]
             else:
                 payload = [command.to_dict() for command in slash_commands]
```
Comment (pythonmcpi) [inline jishaku/features/management.py:246]: The `to_dict()` call on line 244 also needs to be passed the tree as an argument.

**Review by pythonmcpi — CHANGES_REQUESTED** [review]

**Comment by Vioshim** [comment]
Change has been merged in master, jishaku's sync command no longer works with [Master Branch](https://github.com/Rapptz/discord.py/commit/2e2f51fd5c95b3c40fd1c67ae4b08bbeeaeaddce#diff-edc6201f259e50254d3bc72339d8935f50a795fee7f4f0056c31e0641635c493R1740)

**Comment by JDJGInc** [comment]
Can confirm:
```python
May 07 23:24:59 contabo-vps python[1054883]:     ret = await coro(*args, **kwargs)
May 07 23:24:59 contabo-vps python[1054883]:           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
May 07 23:24:59 contabo-vps python[1054883]:   File "/home/jdjg/JDBot/.venv/lib/python3.11/site-packages/jishaku/features/management.py", line 246, in jsk_sync
May 07 23:24:59 contabo-vps python[1054883]:     payload = [command.to_dict() for command in slash_commands]
May 07 23:24:59 contabo-vps python[1054883]:               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
May 07 23:24:59 contabo-vps python[1054883]:   File "/home/jdjg/JDBot/.venv/lib/python3.11/site-packages/jishaku/features/management.py", line 246, in <listcomp>
May 07 23:24:59 contabo-vps python[1054883]:     payload = [command.to_dict() for command in slash_commands]
May 07 23:24:59 contabo-vps python[1054883]:                ^^^^^^^^^^^^^^^^^
May 07 23:24:59 contabo-vps python[1054883]: TypeError: Command.to_dict() missing 1 required positional argument: 'tree'
```

**Comment by Gorialis** [comment]
Because this change still hasn't hit PyPI release, this PR can't be merged until it is ensured that this change doesn't break users who aren't on master branch.

**Comment by Soheab** [comment]
> Because this change still hasn't hit PyPI release, this PR can't be merged until it is ensured that this change doesn't break users who aren't on master branch.

It could handle the error? If this gets merged, it will error for dpy <2.4 users anyways.

**Comment by Gorialis** [comment]
Superceded by #235

**Pull request closed without merging**